Fetch main branch when finalizing

Add a step to fetch the main branch

```
Run git checkout develop
Already on 'develop'
Your branch is up to date with 'origin/develop'.
merge: main - not something we can merge
Error: Process completed with exit code 1.
```

--- a/.github/workflows/finalize-release.yml
+++ b/.github/workflows/finalize-release.yml
@@ -32,6 +32,11 @@ jobs:
             echo "No open pull request found for the release branch."
           fi
 
+      - name: Fetch and checkout main branch
+        run: |
+          git fetch origin main
+          git checkout main
+
       - name: Merge main into develop
         run: |
           git checkout develop

